File: java/07 - Demo 3 - Refactoring/src/main/java/decomposeconditional/TireChange.java
package decomposeconditional;

import java.util.Calendar;

public class TireChange {

	private static final int WINTER_START  = Calendar.DECEMBER;
	private static final int WINTER_END = Calendar.FEBRUARY;
	private int quantity;
	private int winterrate;
	private int winterservicecharge;
	private int summerrate;
	
	public TireChange(int quantity, int winterrate, int winterservicecharge, int summerrate) {
		super();
		this.quantity = quantity;
		this.winterrate = winterrate;
		this.winterservicecharge = winterservicecharge;
		this.summerrate = summerrate;
	}

	int charge(int monthOfYear) {
		int charge;
		if (monthOfYear >= WINTER_START || monthOfYear <= WINTER_END) {
			charge = quantity * winterrate + winterservicecharge;
		} else {
			charge = quantity * summerrate;
		}
		return charge;
	}
}

File: csharp/07 - Demo 3 - Refactoring/Refactoring/DecomposeConditional/TireChange.cs
namespace Refactoring.DecomposeConditional
{
    public class TireChange
    {
        private static readonly int WINTER_START = 12;
        private static readonly int WINTER_END = 2;
        private int _quantity;
        private int _winterrate;
        private int _winterservicecharge;
        private int _summerrate;

        public TireChange(int quantity, int winterrate, int winterservicecharge, int summerrate)
        {
            _quantity = quantity;
            _winterrate = winterrate;
            _winterservicecharge = winterservicecharge;
            _summerrate = summerrate;
        }

        public int Charge(int monthOfYear)
        {
            int charge;
            if (monthOfYear >= WINTER_START || monthOfYear <= WINTER_END)
            {
                charge = _quantity * _winterrate + _winterservicecharge;
            }
            else
            {
                charge = _quantity * _summerrate;
            }
            return charge;
        }
    }
}